wayland-backend/src/protocol.rs

```diff
@@ -242,6 +242,35 @@ pub enum WEnum<T> {
     Unknown(u32),
 }
 
+/// Error representing an unknown numeric variant for a [`WEnum`]
+#[derive(Debug, Copy, Clone)]
+pub struct WEnumError {
+    typ: &'static str,
+    value: u32,
+}
+
+impl std::error::Error for WEnumError {}
+
+impl std::fmt::Display for WEnumError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Unknown numeric value {} for enum {}", self.value, self.typ)
+    }
+}
+
+impl<T> WEnum<T> {
+    /// Convert this [`WEnum`] into a result
+    ///
+    /// This can be used to take advantage of the numerous helper methods on [`Result`] if you
+    /// don't plan to handle the unknown case of this enum.
+    #[inline]
+    pub fn into_result(self) -> Result<T, WEnumError> {
```

Comment:
Can't it be done via `From` or `Into` traits?

Reply:
Well, I can add a `impl<T> From<Wenum<T>> for Result<T, WEnumError>` as well, but I believe that `.into_result()` is much more practical for use in chaining operations than `.into::<Result<T, WEnumError>>()`.